### PR TITLE
Update sample links and remove outdated content

### PR DESCRIPTION
**Pull request by moarychan**
Fixes https://github.com/Azure-Samples/azure-spring-boot-samples/issues/398

- Link to the latest Spring samples.
- Remove the unused file *eng/bomgenerator/validateAzureSpringBootBom.cmd* and configuration since the name `azure-spring-boot-bom` has already changed to `spring-cloud-azure-dependencies`.

### CODE REVIEW

File: eng/bomgenerator/validateAzureSpringBootBom.cmd [hunk 1]
```diff
@@ -1,3 +0,0 @@
-set "bomPomFileLocation=%~dp0..\..\sdk\boms\azure-spring-boot-bom\pom.xml"
-set "reportFileLocation=%~dp0report.html"
-mvn clean install && mvn exec:java -Dexec.args="-mode=analyze -pomFile=%bomPomFileLocation% -reportFile=%reportFileLocation%" && copy %~dp0report.html azurespringbootbomreport.html
```
Comment (chenrujun) [inline eng/bomgenerator/validateAzureSpringBootBom.cmd:3]: Please update `eng/bomgenerator/validateAzureSpringCloudBom.cmd`. Including the filename and `bomPomFileLocation`.

File: eng/versioning/external_dependencies.txt [hunk 2]
```diff
@@ -395,5 +395,5 @@ resourcemanager_com.jcraft:jsch;0.1.55
 # sdk\storage\azure-storage-blob-cryptography\pom.xml
 storage_com.microsoft.azure:azure-storage;8.4.0
 
-# sdk\spring\azure-spring-boot\pom.xml
+# sdk\appconfiguration\azure-spring-cloud-test-appconfiguration-config\pom.xml
```
Comment (chenrujun) [inline eng/versioning/external_dependencies.txt:398]: Why add this one?

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: eng/versioning/external_dependencies.txt (hunk 2 above)
Comment (moarychan) [reply eng/versioning/external_dependencies.txt:398]: Update the context of the note, including the outdated word 'azure-spring-boot', and no additions.

File: eng/bomgenerator/validateAzureSpringBootBom.cmd (hunk 1 above)
Comment (moarychan) [reply eng/bomgenerator/validateAzureSpringBootBom.cmd:3]: There's already https://github.com/Azure/azure-sdk-for-java/blob/main/eng/bomgenerator/validateAzureSpringCloudBom.cmd, just removed the unused one, but seems this one is not referenced by any file.

Comment (chenrujun) [reply eng/bomgenerator/validateAzureSpringBootBom.cmd:3]: > There's already https://github.com/Azure/azure-sdk-for-java/blob/main/eng/bomgenerator/validateAzureSpringCloudBom.cmd,

I mean do like this: 
image

>  just removed the unused one, but seems this one is not referenced by any file.

It's a script file, maybe it should be executed manually.

**Review by chenrujun — APPROVED** [review]

File: eng/bomgenerator/validateAzureSpringBootBom.cmd (hunk 1 above)
Comment (moarychan) [reply eng/bomgenerator/validateAzureSpringBootBom.cmd:3]: Hi @pallavit, please note this file name and pom file path should be updated.

**Pull request merged**